frontend/src/components/custom-tools/list-of-tools/ListOfTools.jsx

```diff
@@ -1,335 +1,115 @@
 import { PlusOutlined } from "@ant-design/icons";
-import { useEffect, useState } from "react";
-
-import { useAxiosPrivate } from "../../../hooks/useAxiosPrivate";
-import { useAlertStore } from "../../../store/alert-store";
-import { useSessionStore } from "../../../store/session-store";
-import { CustomButton } from "../../widgets/custom-button/CustomButton";
-import { AddCustomToolFormModal } from "../add-custom-tool-form-modal/AddCustomToolFormModal";
-import { ViewTools } from "../view-tools/ViewTools";
-import "./ListOfTools.css";
-import { useExceptionHandler } from "../../../hooks/useExceptionHandler";
-import { ToolNavBar } from "../../navigations/tool-nav-bar/ToolNavBar";
-import { SharePermission } from "../../widgets/share-permission/SharePermission";
-import usePostHogEvents from "../../../hooks/usePostHogEvents.js";
-
-let OnboardMessagesModal;
-let slides;
-try {
-  OnboardMessagesModal =
-    require("../../../plugins/onboarding-messages/OnboardMessagesModal.jsx").OnboardMessagesModal;
-  slides =
-    require("../../../plugins/onboarding-messages/login-slides.jsx").LoginSlides;
-} catch (err) {
-  OnboardMessagesModal = null;
-  slides = [];
-}
+import { useCallback, useMemo } from "react";
+import usePostHogEvents from "../../../hooks/usePostHogEvents";
+import { useListManager } from "../../../hooks/useListManager";
+import ListOfToolsModal from "./ListOfToolsModal";
+import { ListView } from "../../view-projects/ListView";
 
 function ListOfTools() {
-  const [isListLoading, setIsListLoading] = useState(false);
-  const [openAddTool, setOpenAddTool] = useState(false);
-  const [editItem, setEditItem] = useState(null);
-  const { sessionDetails } = useSessionStore();
-  const { loginOnboardingMessage } = sessionDetails;
   const { setPostHogCustomEvent } = usePostHogEvents();
 
-  const { setAlertDetails } = useAlertStore();
-  const axiosPrivate = useAxiosPrivate();
+  const getBaseUrl = (sessionDetails) =>
```

Comment:
getBaseUrl naming is bit miss leading since we provide more than baseUrl also can we move this function to static functions where we actually provide base url (till org Id)  when we pass sessionDetails?